scripts/make_width_table.py

```diff
@@ -0,0 +1,50 @@
+import sys
+from os.path import basename, dirname, join
+from typing import Iterable, Tuple
+
+from wcwidth import wcwidth
+
+
+def make_width_table() -> Iterable[Tuple[int, int, int]]:
+    start_codepoint = -1
+    end_codepoint = -1
+    range_width = -2
+    for codepoint in range(0, sys.maxunicode + 1):
+        width = wcwidth(chr(codepoint))
+        if width == 1:
+            continue
+        if start_codepoint < 0:
+            start_codepoint = codepoint
+            range_width = width
+        elif width != range_width or codepoint != end_codepoint + 1:
+            yield (start_codepoint, end_codepoint, range_width)
+            start_codepoint = codepoint
+            range_width = width
+        end_codepoint = codepoint
+    if start_codepoint >= 0:
+        yield (start_codepoint, end_codepoint, range_width)
+
+
+def main() -> None:
+    table_path = join(dirname(__file__), "..", "src", "black", "_width_table.py")
+    with open(table_path, "w") as f:
+        f.write(
+            f"""# Generated by {basename(__file__)}
+import sys
```

Comment:
The output should include the version of relevant tools that was used to generate this. I think that can include:

- wcwidth
- The Unicode version that wcwidth uses
- The Python version, if wcwidth relies on the Unicode DB that comes bundled with Python